clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp

```diff
@@ -11,32 +11,24 @@
 namespace clang {
 namespace dpct {
 
-std::unordered_map<std::string, std::string> APIMapping::EntryMap;
+llvm::DenseMap<llvm::StringRef, llvm::StringRef> APIMapping::EntryMap;
 
-void APIMapping::registerEntry(const std::string &Name,
-                               const std::string &Description) {
-  if (Description == "NA")
-    return;
-  EntryMap[Name] = Description;
+void APIMapping::registerEntry(const llvm::StringRef Name,
+                               const llvm::StringRef SourceCode) {
+  EntryMap[Name] = SourceCode;
 }
 
 void APIMapping::initEntryMap() {
-  static const std::string MultiStr =
-      "\nThere are multi kinds of migrations for this API with different "
-      "migration options,\nsuggest to use the tool to migrate a API usage "
-      "code to see more detail of the migration.";
-#define ENTRY(INTERFACENAME, APINAME, VALUE, FLAG, TARGET, COMMENT, MAPPING)   \
-  registerEntry(#APINAME, MAPPING);
-#define ENTRY_MEMBER_FUNCTION(OBJNAME, INTERFACENAME, APINAME, VALUE, FLAG,    \
-                              TARGET, COMMENT, MAPPING)                        \
-  registerEntry(#OBJNAME "." #APINAME, MAPPING);
-#define MAP_SYCL(MAPPING) MAPPING
-#define MAP_DPCT(MAPPING) MAPPING
-#define MAP_MULTI(MAPPING) MAPPING + MultiStr
-#define MAP_DESC(MAPPING) MAPPING
-#include "../APINames.inc"
-#undef ENTRY_MEMBER_FUNCTION
-#undef ENTRY
+#include "APIMappingRegister.def"
+}
+
+const llvm::StringRef APIMapping::getAPISourceCode(const llvm::StringRef Key) {
```

Comment:
Please remove const qualifier in parameter, `StringRef` do not need this. `StringRef` is a constant view of a string.